codec-http2/src/main/java/io/netty/handler/codec/http2/Http2FrameCodec.java

```diff
@@ -312,6 +315,14 @@ private void increaseInitialConnectionWindow(int deltaBytes) throws Http2Excepti
 
     final boolean consumeBytes(int streamId, int bytes) throws Http2Exception {
         Http2Stream stream = connection().stream(streamId);
+        // upgraded requests are ineligible for stream control
+        if (streamId == Http2CodecUtil.HTTP_UPGRADE_STREAM_ID) {
+            Boolean upgraded = stream.getProperty(upgradeKey);
+            if (upgraded != null && upgraded) {
```

Comment:
one nit: I think you could shortcut this via `Boolean.TRUE.equals(upgrade)`.

Reply:
@normanmaurer I'd rather not depend on escape analysis to avoid the boxing allocation, but I can change it if you feel strongly.

Reply:
@mosesn I am not sure why any escape analysis would be needed here.. Can you explain to me why you think this would be the case ?

`upgraded` will be a `Boolean` anyway so I am not sure what you refer to.

Reply:
@normanmaurer that's an excellent point.  I'll change it.